fs/operations.go

```diff
@@ -1303,3 +1364,24 @@ func MoveFile(fdst Fs, fsrc Fs, dstFileName string, srcFileName string) (err err
 func CopyFile(fdst Fs, fsrc Fs, dstFileName string, srcFileName string) (err error) {
 	return moveOrCopyFile(fdst, fsrc, dstFileName, srcFileName, true)
 }
+
+func (j jsonPrinter) Start(w io.Writer) {
+	syncFprintf(w, "[")
+}
+
+func (j jsonPrinter) End(w io.Writer) {
+	syncFprintf(w, "]")
+}
+
+func (j *jsonPrinter) Fprintf(w io.Writer, skipSeperator bool, format string, a ...interface{}) {
+	// Lets skip the seperator when calling the function itself
+	if j.seperatorRequired && !skipSeperator {
+		j.Fprintf(w, true, ",")
+	}
+	// Set the seperator before locking so parallel things don't get in the way of the lock
+	j.seperatorRequired = true
+
+	j.mu.Lock()
```

Comment:
Test output with lock at the top of the method. Only one line is printed and then rclone locks up

```
rclone.exe ls gd_c:"/MEDIA/YOUTUBE/NoCopyrightSounds (UC_aEa8K-EOJ3D6gOs7HcyNg)/3wjb3HtD32c" --json
[{"Size":   149733,"Remote":"Cartoon feat. Jüri Pootsmann - I Remember U [NCS Official Video]-3wjb3HtD32c.jpg"}
```

Test output with lock in the location shown above. This completes and I can parse the output with jq

```
rclone.exe ls gd_c:"/MEDIA/YOUTUBE/NoCopyrightSounds (UC_aEa8K-EOJ3D6gOs7HcyNg)/3wjb3HtD32c" --json | jq .
[
  {
    "Size": 149733,
    "Remote": "Cartoon feat. Jüri Pootsmann - I Remember U [NCS Official Video]-3wjb3HtD32c.jpg"
  },
  {
    "Size": 34936,
    "Remote": "Cartoon feat. Jüri Pootsmann - I Remember U [NCS Official Video]-3wjb3HtD32c.info.json"
  },
  {
    "Size": 1360,
    "Remote": "Cartoon feat. Jüri Pootsmann - I Remember U [NCS Official Video]-3wjb3HtD32c.annotations.xml"
  },
  {
    "Size": 2308,
    "Remote": "Cartoon feat. Jüri Pootsmann - I Remember U [NCS Official Video]-3wjb3HtD32c.description"
  },
  {
    "Size": 54577921,
    "Remote": "Cartoon feat. Jüri Pootsmann - I Remember U [NCS Official Video]-3wjb3HtD32c.mkv"
  }
]
```